app/android/runtime_client/src/org/xwalk/app/runtime/extension/ReflectionHelper.java

```diff
@@ -0,0 +1,275 @@
+        return members;
+    }
+
+    Boolean hasMethod(String name) {
+        if (!members.containsKey(name)) return false;
+        MemberInfo mInfo = members.get(name);
+        return mInfo.type == MemberType.JS_METHOD;
+    }
+
+    Boolean hasProperty(String name) {
+        if (!members.containsKey(name)) return false;
+        MemberInfo mInfo = members.get(name);
+        return mInfo.type == MemberType.JS_PROPERTY;
+    }
+
+    /*
+     * Use case: construct Java object array from JSON array which is passed by JS
+     * 1. restore original Java object in the array
+     * 2. if the parameter is a callbackID, then combine the instanceID with it
+     */
+    public static Object[] getArgsFromJson(int instanceID, Method m, JSONArray args) {
+        Class<?>[] pTypes = m.getParameterTypes();
+        Object[] oArgs = new Object[pTypes.length];
+        Annotation[][] anns = m.getParameterAnnotations();
+        for (int i = 0; i < pTypes.length; ++i) {
+            try {
+                Class<?> p = pTypes[i];
+                // Identify the callback parameters, JSONObject type with @JsCallback annotation.
+                if (p.equals(JSONObject.class)) {
+                    if (anns[i].length > 0 && anns[i][0] instanceof JsCallback) {
+                        JSONObject callbackInfo = (JSONObject)(args.get(i));
+
+                        // Add extension instance ID to the callbackInfo for message back.
+                        callbackInfo.put("instanceID", instanceID);
+                        oArgs[i] = callbackInfo;
+                    } else {
+                        // For other JSONObject arguments.
+                        oArgs[i] = (JSONObject)(args.get(i));
+                    }
+                } else {
+                    // TODO: check if this is enough for other types.
+                    oArgs[i] = args.get(i);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        return oArgs;
+    }
+
+    public static boolean isSerializable(Object obj) {
+        Class<?> clz = obj.getClass();
+
+        if (clz.isPrimitive() ||
+           primitives.contains(clz) ||
+           obj instanceof String ||
+           obj instanceof Map ||
+           obj instanceof JSONArray ||
+           obj instanceof JSONObject) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static Object toSerializableObject(Object obj) {
+        if (isSerializable(obj)) return obj;
+        if (obj.getClass().isArray()) {
+            JSONArray result = new JSONArray();
+            Object [] arr = (Object[]) obj;
+            for (int i = 0; i < arr.length; ++i) {
+                try {
+                    result.put(i, toSerializableObject(arr[i]));
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+            return result;
+        } else {
```

Comment:
The else block {} indent is not needed because you will return any case in your above if condition.